Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -95,13 +96,50 @@ private void OnActivateInWorld(EntityUid uid, AbsorbentComponent component, User
 
     private void OnAfterInteract(EntityUid uid, AbsorbentComponent component, AfterInteractEvent args)
     {
-        if (!args.CanReach || args.Handled || args.Target == null)
+        if (!args.CanReach || args.Handled)
             return;
 
-        Mop(args.User, args.Target.Value, args.Used, component);
+        if (args.Target == null)
+        {
+            SpillMop(args.User, args.ClickLocation, args.Used, component);
+        }
+        else
+        {
+            Mop(args.User, args.Target.Value, args.Used, component);
+        }
+
         args.Handled = true;
     }
 
+    public void SpillMop(EntityUid user, EntityCoordinates clickLocation, EntityUid used, AbsorbentComponent component)
+    {
+        if (!_solutionContainerSystem.TryGetSolution(used, AbsorbentComponent.SolutionName, out var absorberSoln))
+        {
+            return;
+        }
+
+        if (TryComp<UseDelayComponent>(used, out var useDelay)
+            && _useDelay.IsDelayed((used, useDelay)))
+            return;
+
+        var solution = absorberSoln.Value.Comp.Solution;
+        var spillAmount = component.PickupAmount > solution.Volume ? solution.Volume : component.PickupAmount;
+
+        // spilling the entire 50u mop in one go felt weird, cap to 10u
+        spillAmount = spillAmount > 10 ? 10 : spillAmount;
```

Comment:
How about a percentage of the mop's capacity

Reply:
could work, 20% with a minimum of 1u?

Reply:
Sounds good to me

Reply:
mop contents c:
c > 5u : spill 20%
5u > c > 1u : spill 1u
1u > c : spill c

Reply:
behaves kinda funny at the end when theres very little left, water puddles evaporate fast but it makes a funny noise when you click it fast and i can see a clown or janitor having fun slapping the mop around